stl/inc/format

```diff
@@ -2859,101 +2862,107 @@ _NODISCARD auto make_wformat_args(const _Args&... _Vals) {
     return _Format_arg_store<wformat_context, _Args...>{_Vals...};
 }
 
-template <class _Out, class _CharT>
-using format_args_t = basic_format_args<basic_format_context<_Out, _CharT>>;
-
 template <output_iterator<const char&> _OutputIt>
-_OutputIt vformat_to(
-    _OutputIt _Out, const string_view _Fmt, const format_args_t<type_identity_t<_OutputIt>, char> _Args) {
-    _Format_handler<_OutputIt, char, basic_format_context<_OutputIt, char>> _Handler(_STD move(_Out), _Fmt, _Args);
-    _Parse_format_string(_Fmt, _Handler);
-    return _Handler._Ctx.out();
+_OutputIt vformat_to(_OutputIt _Out, const string_view _Fmt, const format_args _Args) {
+    if constexpr (is_same_v<_OutputIt, _Fmt_it>) {
+        _Format_handler<char> _Handler(_Out, _Fmt, _Args, _Lazy_locale{});
+        _Parse_format_string(_Fmt, _Handler);
+        return _Out;
+    } else {
+        _Fmt_iterator_buffer<_OutputIt, char> _Buf(_STD move(_Out));
+        _Format_handler<char> _Handler(_Fmt_it{_Buf}, _Fmt, _Args, _Lazy_locale{});
+        _Parse_format_string(_Fmt, _Handler);
+        return _Buf._Out();
+    }
 }
 
 template <output_iterator<const wchar_t&> _OutputIt>
-_OutputIt vformat_to(
-    _OutputIt _Out, const wstring_view _Fmt, const format_args_t<type_identity_t<_OutputIt>, wchar_t> _Args) {
-    _Format_handler<_OutputIt, wchar_t, basic_format_context<_OutputIt, wchar_t>> _Handler(
-        _STD move(_Out), _Fmt, _Args);
-    _Parse_format_string(_Fmt, _Handler);
-    return _Handler._Ctx.out();
+_OutputIt vformat_to(_OutputIt _Out, const wstring_view _Fmt, const wformat_args _Args) {
+    if constexpr (is_same_v<_OutputIt, _Fmt_wit>) {
+        _Format_handler<wchar_t> _Handler(_Out, _Fmt, _Args, _Lazy_locale{});

```

Comment:
you can leave off the locale here, there's a constructor that default constructs it.

Reply:
Done.